[AOTInductor] Add non-default device test

khabinov opened the pull request:
Differential Revision: D49604597



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

chenyang78 reviewed (approved):
There is some lint issue. Otherwise, LGTM. Thanks!

khabinov committed:
[AOTInductor] Add non-default device test (#110024)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/110024

Reviewed By: chenyang78

Differential Revision: D49604597
